dirsrvtests/tests/suites/replication/changelog_encryption_test.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import pytest
+import os
+import time
+from lib389._constants import DEFAULT_SUFFIX,  DN_CHANGELOG,  DN_USERROOT_LDBM
+from lib389.topologies import topology_m1c1 as topo
+from lib389.dseldif import DSEldif
+from lib389.utils import ds_supports_new_changelog
+from lib389.replica import Replicas
+
+DEBUGGING = os.getenv("DEBUGGING", default=False)
+if DEBUGGING:
+    logging.getLogger(__name__).setLevel(logging.DEBUG)
+else:
+    logging.getLogger(__name__).setLevel(logging.INFO)
+log = logging.getLogger(__name__)
+
+
+def test_cl_encryption_setup_process(topo):
+    """Take an already working replication deployment, and setup changelog
+    encryption
+
+    :id: 1a1b7d29-69f5-4f0e-91c4-e7f66140ff17
+    :setup: Master Instance, Consumer Instance
+    :steps:
+        1. Enable TLS for the server
+        2. Export changelog
+        3. Enable changelog encryption
+        4. Import changelog
+        5. Verify replication is still working
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+    """
+
+    supplier = topo.ms['master1']
+    consumer = topo.cs['consumer1']
+
+    # Enable TLS
+    log.info('Enable TLS ...')
+    supplier.enable_tls()
+    consumer.enable_tls()
+
+    # Export changelog
+    log.info('Export changelog ...')
+    replicas = Replicas(supplier)
+    replica = replicas.get(DEFAULT_SUFFIX)
+    replica.begin_task_cl2ldif()
```

Comment:
Is there a task.wait()  you could use here?

Reply:
For replica tasks there is "task_finished()" so I used that...

Reply:
Fixed